enterprise/frontend/src/metabase-enterprise/advanced_permissions/components/RoleAttributeMappingModal/RoleAttributeMappingModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect } from "react";
+import { withRouter } from "react-router";
+import { useSelector } from "react-redux";
+import { push } from "react-router-redux";
+import { useDispatch } from "metabase/lib/redux";
+import { fetchAttributes } from "metabase-enterprise/sandboxes/actions";
+import { getAttributes } from "metabase-enterprise/sandboxes/selectors";
+import Databases from "metabase/entities/databases";
+import { State } from "metabase-types/store";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper/LoadingAndErrorWrapper";
+import { RoleAttributeMappingModalParams } from "metabase-enterprise/advanced_permissions/types";
+import { getParentPath } from "metabase/hoc/ModalRoute";
+import RoleAttributeMappingModalView from "./RoleAttributeMappingModalView";
+
+interface RoleAttributeMappingModalProps {
+  params: RoleAttributeMappingModalParams;
+  route: {
+    path: string;
+  };
+}
+
+const RoleAttributeMappingModal = ({
+  params,
+  route,
+}: RoleAttributeMappingModalProps) => {
+  const databaseId = parseInt(params.impersonatedDatabaseId);
+  const attributes = useSelector(getAttributes);
+
+  const database = useSelector((state: State) =>
+    Databases.selectors.getObject(state, { entityId: databaseId }),
+  );
+
+  const dispatch = useDispatch();
+
+  const handleSave = useCallback(() => {
+    console.log("TODO: save");
+  }, []);
+
+  const handleCancel = useCallback(() => {
+    dispatch(push(getParentPath(route, location)));
+  }, [dispatch, route]);
+
+  useEffect(() => {
+    dispatch(fetchAttributes());
+  }, [dispatch]);
```

Comment:
Is this only supposed to run when the component mounts? If so I think it would be better to use `useMount` from `react-use`. It has no dependency array and is more semantic. I have an example of this [here](https://github.com/metabase/metabase/pull/30189/files#diff-9c3174983cdd8b3b488b50f9d2f094016e0f20bfc9fd7fcdbc8489291e30ccd6R23) in my recent PR for dashboard tabs.